lib/Test2/Compare/Negatable.pm

```diff
@@ -0,0 +1,119 @@
+package Test2::Compare::Negatable;
+use strict;
+use warnings;
+
+require overload;
+require Test2::Util::HashBase;
+
+sub import {
+    my ($pkg, $file, $line) = caller;
+
+    my $sub = eval <<"    EOT" or die $@;
+package $pkg;
+#line $line "$file"
+sub { overload->import('!' => 'clone_negate', fallback => 1); Test2::Util::HashBase->import('negate')}
```

Comment:
The current name is not idea, but I am not sure I want to use 'role' in the name either. Lets wait a bit and see if we come up with something better,